### DISCUSSION TITLE
Customize datagrid editing pages actions

### DISCUSSION THREAD

**Discussion by michele-bettin** [Megabit/Blazorise#5057] [General]
Hi guys, thank you really much for creating Blasorise it is awesome...
I'm using the first datagrid on that link https://blazorise.com/docs/extensions/datagrid/features/editing and I was wondering if there was a way to firstly add a kind of callback for the modify > Save button, New > Save button and delete button to actually create, update and delete the selected item on my db and furthermore if there was a way to customize the edit page to add some field that I'm not displaying on the datagrid but are required from the Model...
Thanks you for the answer! <3

**Comment by David-Moreira** [comment]
Hello,

Please try these
![image](https://github.com/Megabit/Blazorise/assets/22283161/6ede0f2b-977a-4391-88e7-b4b4f867f519)
![image](https://github.com/Megabit/Blazorise/assets/22283161/e64c79e3-da01-4825-87bf-818ce9f397c1)
![image](https://github.com/Megabit/Blazorise/assets/22283161/a186205e-cf03-4762-a8a7-6429654fa46a)

Let us know if this helps.

**Reply by michele-bettin** [reply]
thank you very much, it works!
while for inserting a new record, if there are fields in the data model that are not shown in the table and therefore do not appear in the entry form, what should I do?
many thanks in advance

**Reply by David-Moreira** [reply]
Sorry I don't follow, can you rephrase your question, please?

Thanks!

**Reply by michele-bettin** [reply]
My problem is that I have a Model that represents a User. A User has 3 properties/fields (Username, Email and Password) but in the table where I list all the users I show only Email and Password. But the administrator that has access to this table would like also to create new users (it has no sense but it is for didactical purpose) and in the form he has to insert also the password (with validation), but the form only shows 2 fields...

**Reply by David-Moreira** [reply]
You should be able to use these parameters.
![image](https://github.com/Megabit/Blazorise/assets/22283161/141244ac-75cf-4171-b720-13630677d310)

**Reply by michele-bettin** [reply]
Man, you saved my life!!! Thank you so much! I'm happy to have found someone who helps without despising people with less experience, I hope you will answer me in the future too if I need it!
Michele
